### PR TITLE
Fixing crash

### LINKED ISSUE #1323: App crashing while uploading certain pictures.
**Summary:** 
I tried to upload one picture the app crashed. But it's not happening with every picture that I try to upload. Happens with picture located in content://com.android.providers.downloads.documents/document/

**Steps to reproduce:** 
Open app and click on upload picture and try to upload picture(Does not happen with every picture).

**Add System logs:**
```
03-17 03:06:15.693 748-748/fr.free.nrw.commons.debug E/AndroidRuntime: FATAL EXCEPTION: main
Process: fr.free.nrw.commons.debug, PID: 748
java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons.debug/fr.free.nrw.commons.upload.ShareActivity}: java.lang.NullPointerException: Attempt to get length of null array
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2659)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2724)
at android.app.ActivityThread.-wrap12(ActivityThread.java)
 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1473)
at android.os.Handler.dispatchMessage(Handler.java:102)
 at android.os.Looper.loop(Looper.java:154)
 at android.app.ActivityThread.main(ActivityThread.java:6123)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)
Caused by: java.lang.NullPointerException: Attempt to get length of null array
at fr.free.nrw.commons.upload.ShareActivity.findOtherImages(ShareActivity.java:537)
at fr.free.nrw.commons.upload.ShareActivity.getFileMetadata(ShareActivity.java:519)
at fr.free.nrw.commons.upload.ShareActivity.performPreUploadProcessingOfFile(ShareActivity.java:398)
at fr.free.nrw.commons.upload.ShareActivity.onCreate(ShareActivity.java:298)
at android.app.Activity.performCreate(Activity.java:6672)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1140)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2612)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2724) 
at android.app.ActivityThread.-wrap12(ActivityThread.java) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1473) 
at android.os.Handler.dispatchMessage(Handler.java:102) 
at android.os.Looper.loop(Looper.java:154) 
 at android.app.ActivityThread.main(ActivityThread.java:6123) 
 at java.lang.reflect.Method.invoke(Native Method) 
 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867) 
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757) 
03-17 03:06:16.375 748-1001/fr.free.nrw.commons.debug D/ExistingFileAsync: File already exists in Commons: false


```

**Expected behavior:** 
It should not crash.

**Observed behavior:** 
It crashed.
device-2018-03-16-194758

**Device and Android version:** 
Moto G4 plus 
Android 7.0.0
Xiaomi
 8.1.0

 **Commons app version:** 
2.6.7

**Would you like to work on the issue?**
Yes i would like to work on this issue.

### PR DESCRIPTION
**Pull request by Mansi449**
## Description

Fixes #1323 

Made changes according to the comment: 
https://github.com/commons-app/apps-android-commons/issues/1323#issuecomment-373899479

## Tests performed

Tested on Motorola Moto G(5) (Android 7.0, API Level 24)

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/upload/FileUtils.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public static String getPath(Context context, Uri uri) {
 
                 final String id = DocumentsContract.getDocumentId(uri);
                 final Uri contentUri = ContentUris.withAppendedId(
-                        Uri.parse("content://downloads/public_downloads"), Long.valueOf(id));
+                        Uri.parse("content://downloads/document"), Long.valueOf(id));
```
Comment (psh) [inline app/src/main/java/fr/free/nrw/commons/upload/FileUtils.java]: Can you explain why this change is relevant to sharing the app from the _About_ screen?

If it's related to fixing a different issue, I'd suggest rolling it into its own PR and not including it here.

Comment (Mansi449) [reply app/src/main/java/fr/free/nrw/commons/upload/FileUtils.java]: Yes it is related to a different issue. I would close the PR immediately. Sorry for the inconvenience.
